### PR TITLE
[mcs/build] Use make functions instead of calling a shell

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

File: mcs/build/rules.make [hunk 1]
```diff
@@ -175,18 +174,16 @@ do-all-aot:
 # be able to evaluate the .dylibs to make
 ifneq ("$(wildcard $(topdir)/class/lib/$(PROFILE))","")
 
-AOT_PROFILE_ASSEMBLIES_CMD = cd $(topdir)/class/lib/$(PROFILE)/ && find . | grep -E '(dll|exe)$$' | grep -v -E 'bare|plaincore|secxml|Facades|ilasm' | sed 's:\./::g' | tr '\n' ' '
-AOT_PROFILE_ASSEMBLIES_CMD_SAFE = $(AOT_PROFILE_ASSEMBLIES_CMD) || true
-AOT_PROFILE_ASSEMBLIES = $(shell $(AOT_PROFILE_ASSEMBLIES_CMD_SAFE))
+AOT_PROFILE_ASSEMBLIES := $(sort $(patsubst .//%,%,$(filter-out %bare% %plaincore% %secxml% %Facades% %ilasm%,$(filter %.dll %.exe,$(wildcard $(topdir)/class/lib/$(PROFILE)/*)))))
```
Comment (akoeplinger) [inline mcs/build/rules.make]: what is the `patsubst .//` here used for?

Comment (ludovic-henry) [inline mcs/build/rules.make]: the output from `$(filter-out %bare% %plaincore% %secxml% %Facades% %ilasm%,$(filter %.dll %.exe,$(wildcard $(topdir)/class/lib/$(PROFILE)/*)))` might start with `.//` so we simply remove it.

Comment (akoeplinger) [inline mcs/build/rules.make]: when would this be the case?

Comment (akoeplinger) [inline mcs/build/rules.make]: when would this be the case?

**Pull request merged**